File: iOS_Crypto_Module/Algorithm/AES_256_ECB/NSAESString.m
//
//  NSAESString.m
//  密语输入法
//
//  Created by macsjh on 15/9/24.
//  Copyright © 2015年 macsjh. All rights reserved.
//

#import "NSAESString.h"

@implementation NSAESString:NSObject

+(NSString *) aes256_encrypt:(NSString *)PlainText :(NSString*)Key
{
	const char *cstr = [PlainText cStringUsingEncoding:NSUTF8StringEncoding];
	NSData *data = [NSData dataWithBytes:cstr length:PlainText.length];
	//对数据进行加密
	NSData *result = [data aes256_encrypt:Key];
 
	//转换为2进制字符串
	if (result && result.length > 0) {
		
		Byte *datas = (Byte*)[result bytes];
		NSMutableString *output = [NSMutableString stringWithCapacity:result.length * 2];
		for(int i = 0; i < result.length; i++){
			[output appendFormat:@"%02x", datas[i]];
		}
		return output;
	}
	return nil;
}

+(NSString *) aes256_decrypt:(NSString *)CipherText :(NSString *)key
{
	//转换为2进制Data
	NSMutableData *data = [NSMutableData dataWithCapacity:CipherText.length / 2];
	unsigned char whole_byte;
	char byte_chars[3] = {'\0','\0','\0'};
	int i;
	for (i=0; i < [CipherText length] / 2; i++) {
		byte_chars[0] = [CipherText characterAtIndex:i*2];
		byte_chars[1] = [CipherText characterAtIndex:i*2+1];
		whole_byte = strtol(byte_chars, NULL, 16);
		[data appendBytes:&whole_byte length:1];
	}
 
	//对数据进行解密
	NSData* result = [data aes256_decrypt:key];
	if (result && result.length > 0) {
		return [[NSString alloc] initWithData:result encoding:NSUTF8StringEncoding];;
	}
	return nil;
}
@end

File: CryptoFramework/Crypto/Algorithm/AES_256_ECB/NSAESString.swift
//
//  NSAESString.swift
//  密语输入法
//
//  Created by macsjh on 15/10/17.
//  Copyright © 2015年 macsjh. All rights reserved.
//

import Foundation
class NSAESString:NSObject
{
	internal static func aes256_encrypt(PlainText:NSString, Key:NSString) -> NSString?
	{
		let cstr:UnsafePointer<Int8> = PlainText.cStringUsingEncoding(NSUTF8StringEncoding)
		let data = NSData(bytes: cstr, length:PlainText.length)
		//对数据进行加密
		let result = data.aes256_encrypt(Key)
		if (result != nil && result!.length > 0 ) {
			return result!.toHexString()
		}
		return nil
	}
	
	internal static func aes256_decrypt(CipherText:String, key:NSString) -> NSString?
	{
		//转换为2进制Data
		let data = CipherText.hexStringToData()
		
		//对数据进行解密
		let result = data!.aes256_decrypt(key);
		if (result != nil && result!.length > 0) {
			return NSString(data:result!, encoding:NSUTF8StringEncoding)
		}
		return nil;
	}
}